### PR TITLE
Grafana dashboards for kube-prometheus

### PR DESCRIPTION
**Pull request by goruha**
## What
* Fetch dashboards from https://github.com/coreos/prometheus-operator/tree/master/helm/grafana/dashboards
* Replace `${DS_PROMETHEUS}` with `prometheus`

## Why
* Files in https://github.com/coreos/prometheus-operator/tree/master/helm/grafana/dashboards have wrong format that can not be imported by https://github.com/helm/charts/tree/master/stable/grafana
* Workaround for https://github.com/grafana/grafana/issues/10786

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+## Write chamber secrets for agent
+kube-prometheus/apply:
+	@mkdir -p ./kube-prometheus
+	@for dashboard_file in $(KUBE_PROMETHEUS_DASHBOARDS) ; do \
+		echo "Fetching kube-prometheus $$dashboard_file version $(KUBE_PROMETHEUS_VERSION)"; \
+		curl -s $(KUBE_PROMETHEUS_BASE_URL)/$$dashboard_file | jq .dashboard | envsubst > ./kube-prometheus/$$dashboard_file ; \
+	done
+
+
+## Write chamber secrets for agent
```
Comment (aknysh) [inline Makefile]: should we update the comments to make them more specific?
and the comments are the same for both Make targets.

**Review by aknysh — COMMENTED** [review]
see comments

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+KUBE_PROMETHEUS_DASHBOARDS += kubernetes-cluster-status-dashboard.json
+KUBE_PROMETHEUS_DASHBOARDS += kubernetes-control-plane-status-dashboard.json
+KUBE_PROMETHEUS_DASHBOARDS += kubernetes-resource-requests-dashboard.json
+KUBE_PROMETHEUS_DASHBOARDS += nodes-dashboard.json
+KUBE_PROMETHEUS_DASHBOARDS += pods-dashboard.json
+KUBE_PROMETHEUS_DASHBOARDS += statefulset-dashboard.json
+
+export DS_PROMETHEUS ?= Prometheus
+
+apply: kube-prometheus/apply nginx/apply
```
Comment (osterman) [inline Makefile]: I think `import` would be a more apropos target name as we are importing and then transforming (but not loading) the dashboards.

**Review by osterman — CHANGES_REQUESTED** [review]
Update to use `README.yaml`

Add references section for chart and helmfile

File: README.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+badges:
+  - name: "Latest Release"
+    image: "https://img.shields.io/github/release/cloudposse/grafana-dashboards.svg"
+    url: "https://github.com/cloudposse/grafana-dashboards/releases"
+  - name: "Slack Community"
+    image: "https://slack.cloudposse.com/badge.svg"
+    url: "https://slack.cloudposse.com"
+
+description: |-
+  Grafana dashboards imported from open source projects and proccessed to be imported with grafana dashboard provisioner.
```
Comment (aknysh) [inline README.yaml]: Don't use `import` two times?
Grafana dashboards taken from open source projects ...

File: README.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+introduction: |-
+  Starting from [Grafana 5.0](http://docs.grafana.org/administration/provisioning/#dashboards) there is provisioning
```
Comment (aknysh) [inline README.yaml]: Starting with [Grafana 5.0](http://docs.grafana.org/administration/provisioning/#dashboards) there is a provisioning system that uses config files to create/update dashboards.

File: README.yaml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  system that uses config files to create\update dashboards.
+
+  [The problem](https://github.com/grafana/grafana/issues/10786) with the provisioner is that it does not support
+  dashboards variables.
+
+  This project is about workaround for the problem.
```
Comment (aknysh) [inline README.yaml]: This project is a workaround for the problem

File: README.yaml [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  It fetch dashboard files and process them replacing `${DS_PROMETHEUS}` with `Prometheus` value and save it.
```
Comment (aknysh) [inline README.yaml]: It fetches dashboard files and processes them replacing `${DS_PROMETHEUS}` with `Prometheus`, and then saves it

File: README.yaml [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+  Processed dashboards files are commited into the repo, so you can provision them by putting the files into grafana
```
Comment (aknysh) [inline README.yaml]: commited. Misspelling of committed :)
https://en.wiktionary.org/wiki/commited

File: README.yaml [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+  provisioner dashboards directory. __Required existed prometheus datasource named `Prometheus`__
+
+  Current supported dashboards:
```
Comment (aknysh) [inline README.yaml]: Currently supported dashboards

File: README.yaml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+  system that uses config files to create\update dashboards.
+
+  [The problem](https://github.com/grafana/grafana/issues/10786) with the provisioner is that it does not support
+  dashboards variables.
+
+  This project is about workaround for the problem.
+  It fetch dashboard files and process them replacing `${DS_PROMETHEUS}` with `Prometheus` value and save it.
+
+  Processed dashboards files are commited into the repo, so you can provision them by putting the files into grafana
+  provisioner dashboards directory. __Required existed prometheus datasource named `Prometheus`__
```
Comment (aknysh) [inline README.yaml]: __Requires existing Prometheus datasource named `Prometheus`__

File: README.yaml [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+  Current supported dashboards:
+  * [kube-prometheus](https://github.com/coreos/prometheus-operator/master/helm/grafana/dashboards/)
+  * [nginx-ingress](https://github.com/kubernetes/ingress-nginx/master/deploy/grafana/dashboards/)
+
+include:
+  - "docs/targets.md"
+
+quickstart: |-
+
+  Config grafana provisioner following [instructions](http://docs.grafana.org/administration/provisioning/#dashboards)
```
Comment (aknysh) [inline README.yaml]: Config Grafana provisioner by following these [instructions]

File: README.yaml [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+  wget https://raw.githubusercontent.com/cloudposse/grafana-dashboards/feature-kube-prometheus-dashboards/kube-prometheus/deployment-dashboard.json /var/lib/grafana/dashboards
+  ```
+
+related:
+  - name: "Grafana"
+    description: "Official grafana chart"
+    url: "https://github.com/helm/charts/tree/master/stable/grafana"
+
+  - name: "Helmfiles"
+    description: "Example provisioning the dashboards with stable grafana chart."
```
Comment (aknysh) [inline README.yaml]: Example of provisioning

File: README.yaml [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+    url: "https://github.com/cloudposse/helmfiles/blob/master/helmfile.d/0400.kube-prometheus.yaml"
+
+  - name: "Monochart"
+    description: "A declarative helm chart for deploying common types of services on Kubernetes"
+    url: "https://github.com/cloudposse/charts/tree/master/incubator/monochart"
+
+references:
+  - name: "Cloud Posse Documentation"
+    description: "Complete documentation for the Cloud Posse solution"
```
Comment (aknysh) [inline README.yaml]: Complete documentation for the Cloud Posse solutions

**Review by aknysh — COMMENTED** [review]
see comments

**Comment by goruha** [comment]
@aknysh - Addressed comments

**Review by aknysh — APPROVED** [review]

**Review by osterman — APPROVED** [review]

**Pull request merged**